tutorials/independence.py

```diff
@@ -313,4 +313,151 @@
 # information.
 # These tests runs like :ref:`any other test<general indep>`.
 #
+########################################################################################
+#Friedman Rafsky Test for Randomness
+# --------------------------------------------
+#
+#This notebook will introduce the usage of the Friedman Rafsky test, a multivariate
+#extension of the Wald-Wolfowitz runs test to test for randomness between two multivariate
+#samples. More specifically, the function tests whether two multivariate samples were
+#independently drawn from the same distribution.
+#
+#The question proposed in 'Multivariate Generalizations of the Wald-Wolfowitz and Smirnov Two-Sample Tests' 
+#is that of how to extend the univariate Wald-Wolfowitz runs test to a multivariate setting.
+#
+#The univariate Wald-Wolfowitz runs test is a non-parametric statistical test that checks a randomness hypothesis
+#for a two-valued data sequence. More specifically, it can be used to test the hypothesis that the elements of a sequence 
+#are mutually independent. For a data sequence with identifiers of two groups,
+#say: $$ X \in \mathbb{R}^n , Y \in \mathbb{R}^m $$ we begin by sorting the combined data set 
+#:math: `W \in \mathbb{R}^{m+n}` in numerical ascending order. The number of runs is then defined by the number 
+#of maximal, non-empty segments of the sequence consisting of adjacent and equal elements. 
+#So if we designate every :math: `X = +,  Y = -` an example assortment of the 15 element long sequence 
+#of both sets could be given as :math: `+++----++---+++` which contains 5 runs, 3 of which positive and 2 of which negative. 
+#By randomly permuting the labels of our data a large number of times, we can compare our true number of runs to that of the 
+#random permutations to determine the test statistic and p-value.
+#
+#For extension to the multivariate case, we determine a neighbor not by proximity in numerical order, 
+#but instead by euclidean distance between each point. The data is then 'sorted' via a calculation of 
+#the minimum spanning tree in which each point is connected to each other point with edge weight equal 
+#to that of the euclidean distance between the two points. The number of runs is then determined by severing 
+#each edge of the MST for which the points connected do not belong to the same family. Similarly to the univariate 
+#case, the labels associated with each node are then permuted randomly a large number of times, we can 
+#compare our number of true runs to the random distribution of runs to determine the multivariate test 
+#statistic and p-value. To further understand, let us consider an exmaple in the 2-dimensional case. 
+#
+#
+########################################################################################
+#
+from IPython.display import Image
```

Comment:
use this instead: https://sublime-and-sphinx-guide.readthedocs.io/en/latest/images.html